src/battle/anim/flying.c

```diff
@@ -23,6 +31,288 @@ void sub_80DB458(struct Sprite *sprite);
+    sprite->pos1.y = GetBattlerSpriteCoord(gAnimBankAttacker, 3);
+    sprite->pos1.x += gBattleAnimArgs[0];
+    sprite->pos1.y += gBattleAnimArgs[1];
+    sprite->data[0] = gBattleAnimArgs[4];
+
+    if (gBattleAnimArgs[6] == 0)
+    {
+        sprite->data[2] = GetBattlerSpriteCoord(gAnimBankTarget, 2);
+        sprite->data[4] = GetBattlerSpriteCoord(gAnimBankTarget, 3);
+    }
+    else
+    {
+        SetAverageBattlerPositions(gAnimBankTarget, 1, &sprite->data[2], &sprite->data[4]);
+    }
+
+    sprite->data[2] = sprite->data[2] + gBattleAnimArgs[2];
+    sprite->data[4] = sprite->data[4] + gBattleAnimArgs[3];
+    sprite->callback = StartTranslateAnimSpriteByDeltas;
+
+    StoreSpriteCallbackInData(sprite, DestroyAnimSprite);
+    SeekSpriteAnim(sprite, gBattleAnimArgs[5]);
+}
+
+void sub_80DA300(struct Sprite *sprite)
+{
+
+    InitAnimSpritePos(sprite, 1);
+    sprite->data[0] = gBattleAnimArgs[2];
+    sprite->data[1] = gBattleAnimArgs[3];
+    sprite->callback = sub_80DA348;
+    gSprites[GetAnimBattlerSpriteId(0)].invisible = 1;
+}
+
+void sub_80DA348(struct Sprite *sprite)
+{
+    if (sprite->data[0] > 0)
+    {
+        sprite->data[0]--;
+    }
+    else
+    {
+        sprite->data[2] += sprite->data[1];
+        sprite->pos2.y -= (sprite->data[2] >> 8);
+    }
+
+    if (sprite->pos1.y + sprite->pos2.y < -32)
+    {
+        DestroyAnimSprite(sprite);
+    }
+}
+
+void sub_80DA38C(struct Sprite *sprite)
+{
+    if (GetBattlerSide(gAnimBankAttacker) != B_SIDE_PLAYER)
+    {
+        sprite->pos1.x = 272;
+        sprite->pos1.y = -32;
+        StartSpriteAffineAnim(sprite, 1);
+    }
+    else
+    {
+        sprite->pos1.x = -32;
+        sprite->pos1.y = -32;
+    }
+
+    sprite->data[0] = gBattleAnimArgs[0];
+    sprite->data[1] = sprite->pos1.x;
+    sprite->data[2] = GetBattlerSpriteCoord(gAnimBankTarget, 2);
+    sprite->data[3] = sprite->pos1.y;
+    sprite->data[4] = GetBattlerSpriteCoord(gAnimBankTarget, 3);
+
+    InitAnimSpriteTranslationDeltas(sprite);
+    sprite->callback = sub_80DA410;
+}
+
+void sub_80DA410(struct Sprite *sprite)
+{
+    sprite->data[0] = 1;
+    TranslateAnimSpriteByDeltas(sprite);
+
+    if (((u16)sprite->data[3] >> 8) > 200)
+    {
+        sprite->pos1.x += sprite->pos2.x;
+        sprite->pos2.x = 0;
+        sprite->data[3] &= 0xFF;
+    }
+
+    if ((u32)(sprite->pos1.x + sprite->pos2.x + 32) > 304 || sprite->pos1.y + sprite->pos2.y > 160)
+    {
+        gSprites[GetAnimBattlerSpriteId(0)].invisible = 0;
+        DestroyAnimSprite(sprite);
+    }
+}
+
+void sub_80DA48C(struct Sprite *sprite)
+{
+    if (sprite->data[0]-- <= 0)
+    {
+        if (sprite->oam.affineMode & 1)
+        {
+            FreeOamMatrix(sprite->oam.matrixNum);
+            sprite->oam.affineMode = 0;
+        }
+        
+        DestroySprite(sprite);
+        gAnimVisualTaskCount--;
+    }
+}
+
 const struct SpriteTemplate gBattleAnimSpriteTemplate_83DA380 =
```

Comment:
Can you move the .rodata above the code?  (.rodata is all of the `const` data). That's how this project is typically organized